lib/dns.ex

```diff
@@ -10,27 +10,29 @@ defmodule DNS do
       iex> DNS.resolve("tungdao.com", :txt)
       {:ok, [['v=spf1 a mx ~all']]}
 
-      iex> DNS.resolve("tungdao.com", :a, {"8.8.8.8", 53})
+      iex> DNS.resolve("tungdao.com", :a, nameservers: [{"8.8.4.4", 53}])
       {:ok, [{1, 1, 1, 1}]}
 
-      iex> DNS.resolve("tungdao.com", :a, {"8.8.8.8", 53}, :tcp)
+      iex> DNS.resolve("tungdao.com", :a, usevc: true)
       {:ok, [{1, 1, 1, 1}]}
 
   """
-  @spec resolve(String.t(), atom, {String.t(), :inet.port()}, :tcp | :udp) ::
-          {atom, :inet.ip()} | {atom, list} | {atom, atom}
-  def resolve(domain, type \\ :a, dns_server \\ {"8.8.8.8", 53}, proto \\ :udp) do
-    case query(domain, type, dns_server, proto).anlist do
-      answers when is_list(answers) and length(answers) > 0 ->
+  @spec resolve(String.t(), atom, list(), timeout()) :: {:ok, list()} | {:error, :inet_res.res_error()}
```

Comment:
It would be nice to have a more explicit/helpful spec for the `opts`. Using just `list` is a bit mysterious.

The `query/4` spec, for example, is using `[:inet_res.res_option()]` for its `opts`.

I'm not 100% sure but I think `nameserver` can be a string instead of a tuple? The spec could make that clear (if it is true).